lib/galaxy/datatypes/tabular.py

```diff
@@ -1161,3 +1161,92 @@ def get_chunk(self, trans, dataset, chunk):
         ck_data_body = re.sub('[ ]+', '\t', ck_data_body)
 
         return dumps({'ck_data': util.unicodify(ck_data_header + "\n" + ck_data_body), 'ck_index': ck_index + 1})
+
+
+class MatrixMarket(TabularData):
+    """
+    The Matrix Market (MM) exchange formats provide a simple mechanism
+    to facilitate the exchange of matrix data. MM coordinate format is
+    suitable for representing sparse matrices. Only nonzero entries need
+    be encoded, and the coordinates of each are given explicitly.
+
+    The tabular file format is defined as follows::
+
+    %%MatrixMarket matrix coordinate real general <--- header line
+    %                                             <--+
+    % comments                                       |-- 0 or more comment lines
+    %                                             <--+
+        M  N  L                                   <--- rows, columns, entries
+        I1  J1  A(I1, J1)                         <--+
+        I2  J2  A(I2, J2)                            |
+        I3  J3  A(I3, J3)                            |-- L lines
+            . . .                                    |
+        IL JL  A(IL, JL)                          <--+
+
+    Indices are 1-based, i.e. A(1,1) is the first element.
+
+    >>> from galaxy.datatypes.sniff import get_test_fname
+    >>> MatrixMarket().sniff( get_test_fname( 'sequence.maf' ) )
+    False
+    >>> MatrixMarket().sniff( get_test_fname( '1.mtx' ) )
+    True
+    >>> MatrixMarket().sniff( get_test_fname( '2.mtx' ) )
+    True
+    """
+    file_ext = "mtx"
+
+    def __init__(self, **kwd):
+        super(MatrixMarket, self).__init__(**kwd)
+
+    def sniff(self, filename):
+        count = 0
+        nonzeros = 0
+        with open(filename, 'r') as handle:
+            lines = handle.readlines()
```

Comment:
This does unbounded reads - this can crash the Galaxy server. Isn't that header line enough to say this is that kind of file?

Reply:
Using the newer sniff_prefix handling (just from ``@build_sniff_from_prefix`` on the class definition), I feel like the following would be a fine sniffer:

```python
def sniff_prefix(self, file_prefix):
    return file_prefix.startswith('%%MatrixMarket matrix coordinate real general')
```

There are advantages to sniff_prefix - it avoids unneeded I/O and keeps memory consumption bounded.

Reply:
I made the change as you suggested @jmchilton 
Thanks 😄